python/semantic_kernel/connectors/ai/hugging_face/services/hf_text_completion.py

```diff
@@ -68,7 +71,7 @@ def __init__(
             "cuda:" + str(device)
             if device >= 0 and torch.cuda.is_available()
             else "cpu"
-        )
+        ) if device is not None else None
```

Comment:
If possible, I would like to understand this behavior better.
Documentation says `Do not use device_map AND device at the same time as they will conflict`.

Let's check different initialization variations:
1. `sk_hf.HuggingFaceTextCompletion('model-id', device=None, pipeline_kwargs=dict(device_map='auto'))` - `device_map` parameter will be used, as expected.
2. `sk_hf.HuggingFaceTextCompletion('model-id', device=1)` - `device` parameter will be used, as expected.
3. `sk_hf.HuggingFaceTextCompletion('model-id', pipeline_kwargs=dict(device_map='auto'))` - since `device` parameter in constructor has default value `-1`, both `device` and `device_map` parameters will be used. Is this expected?
4. `sk_hf.HuggingFaceTextCompletion('model-id', device=1, pipeline_kwargs=dict(device_map='auto'))` - same here, both parameters will be used. Should we handle it in some way or just pass it as it is to pipeline? Or maybe we should leave the same message in our documentation as in HF docs - to avoid passing both parameters at the same time?

Please let me know if I'm missing something. From these variations, I think item 3 should be fixed, if it's possible.